modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNodeObjectFields.spec.tsx

```diff
@@ -0,0 +1,165 @@
+	() => {
+		return {
+			useObjectFolderContext() {
+				return [{}, jest.fn()];
+			},
+		};
+	}
+);
+
+jest.mock('react-flow-renderer', () => {
+	return {
+		useStore() {
+			return {
+				edges: [],
+				nodes: [],
+			};
+		},
+	};
+});
+
+const objectFieldsMock: ObjectFieldNodeRow[] = [
+	{
+		businessType: 'LongInteger',
+		externalReferenceCode: 'objectField1',
+		label: {
+			en_US: 'objectField1',
+		},
+		name: 'objectField1',
+		primaryKey: false,
+		required: false,
+		selected: false,
+	},
+	{
+		businessType: 'LongInteger',
+		externalReferenceCode: 'objectField2',
+		label: {
+			en_US: 'objectField2',
+		},
+		name: 'objectField2',
+		primaryKey: false,
+		required: false,
+		selected: false,
+	},
+	{
+		businessType: 'LongInteger',
+		externalReferenceCode: 'objectField3',
+		label: {
+			en_US: 'objectField3',
+		},
+		name: 'objectField3',
+		primaryKey: false,
+		required: false,
+		selected: false,
+	},
+	{
+		businessType: 'LongInteger',
+		externalReferenceCode: 'objectField4',
+		label: {
+			en_US: 'objectField4',
+		},
+		name: 'objectField4',
+		primaryKey: false,
+		required: false,
+		selected: false,
+	},
+	{
+		businessType: 'LongInteger',
+		externalReferenceCode: 'objectField5',
+		label: {
+			en_US: 'objectField5',
+		},
+		name: 'objectField5',
+		primaryKey: false,
+		required: false,
+		selected: false,
+	},
+	{
+		businessType: 'LongInteger',
+		externalReferenceCode: 'objectField6',
+		label: {
+			en_US: 'objectField6',
+		},
+		name: 'objectField6',
+		primaryKey: false,
+		required: false,
+		selected: false,
+	},
+];
+
+describe('ObjectDefinitionNodeFields Component', () => {
+	it('check if the object field label and business type will be rendered correctly', () => {
+		const {queryByText} = render(
+			<ObjectDefinitionNodeFields
+				defaultLanguageId="en_US"
+				objectFields={[
+					{
+						businessType: 'LongInteger',
+						externalReferenceCode: 'idObjectFieldERC',
+						label: {
+							en_US: 'ID',
+						},
+						name: 'id',
+						primaryKey: true,
+						required: false,
+						selected: false,
+					},
+				]}
+				selectedObjectDefinitionId={0}
+				showAllObjectFields={false}
+			/>
+		);
+
+		expect(queryByText('ID')).toBeVisible();
+		expect(queryByText('long-integer')).toBeVisible();
+	});
+
+	it('check if only 5 object fields will be listed if showAllObjectFields property is false', () => {
+		const {queryByText} = render(
+			<ObjectDefinitionNodeFields
+				defaultLanguageId="en_US"
+				objectFields={objectFieldsMock}
+				selectedObjectDefinitionId={0}
+				showAllObjectFields={false}
+			/>
+		);
+
+		expect(queryByText('objectField1')).toBeVisible();
+		expect(queryByText('objectField2')).toBeVisible();
+		expect(queryByText('objectField3')).toBeVisible();
+		expect(queryByText('objectField4')).toBeVisible();
+		expect(queryByText('objectField5')).toBeVisible();
+		expect(queryByText('objectField6')).toBeNull();
```

Comment:
i think that maybe this isn't the best way to verify that, since the definition of `toBeNull` is:

> .toBeNull() is the same as .toBe(null) but the error messages are a bit nicer. So use .toBeNull() when you want to check that something is null. [x](https://archive.jestjs.io/docs/pt-br/expect#tobenull:~:text=Copy-,.toBeNull(),-.toBeNull()%20is)

i understand what you are doing, but i wonder if there isn't a way on jest to do like `not.toBeVisible` or something more reliable/readable?

Reply:
Hey @rebsilva, When using this `queryByText` function, two possibilities will be returned to us, either the Return element in case of finding the element or null in case of not finding what you are looking for. So in this case using this function it must be validated using toBeNull(). [see](https://testing-library.com/docs/queries/about/#types-of-queries).

I tried to use the `getByText` function but if the element was not found, this function throws an error instead of returning that nothing was found. So instead of creating a treatment for this error, I preferred to use `queryByText` which in this case returns something that could be useful and in practical terms means the same thing.

Reply:
i understand! thanks for explained, i think its fine to use `queryByText` on this case then :blush: